implementations/rust/ockam/ockam_transport_udp/src/puncture/rendezvous_service/messages.rs

```diff
@@ -0,0 +1,48 @@
+use minicbor::{Decode, Encode};
+use ockam_core::{Decodable, Encodable, Encoded, Message, Result};
+
+/// Request type for UDP Puncture Rendezvous service
+#[derive(Encode, Decode, Debug)]
+#[rustfmt::skip]
+pub enum RendezvousRequest {
+    /// Ping service to see if it is reachable and working.
+    #[n(0)] Ping,
+    /// Get my public IP and port
+    #[n(1)] GetMyAddress,
+}
+
+impl Encodable for RendezvousRequest {
+    fn encode(self) -> Result<Encoded> {
+        Ok(minicbor::to_vec(self)?)
+    }
+}
+
+impl Decodable for RendezvousRequest {
+    fn decode(e: &[u8]) -> Result<Self> {
+        Ok(minicbor::decode(e)?)
+    }
+}
+
+impl Message for RendezvousRequest {}
+
+/// Response type for UDP Puncture Rendezvous service
+#[derive(Encode, Decode, Debug)]
+#[rustfmt::skip]
+pub enum RendezvousResponse {
+    #[n(0)] Pong,
+    #[n(1)] GetMyAddress(#[n(0)] String),
```

Comment:
Why not use a proper `Address` here? Is it because we want to eventually remove the transport type from the `Address`?

Reply:
I don't think that's the type it is. `Address` is an Ockam Routing address. This one is a UDP address

Reply:
It's actually `SocketAddr`. So, we could maybe wrap it inside something `Encode`-able as an improvement and use here